src/dict.js

```diff
@@ -15,10 +15,40 @@ Sk.builtin.dict = function dict (L) {
         L = [];
     }
 
+    /**
+     * the internal representation is a combination of a javascript hash map (this.entries)
+     * and a series of buckets (this.buckets) to account for collisions of the skulpt/python hash function
+     * 
+     * this.entries is used as a fast path for items where the key is a python str
+     * As well as to preserve insertion order for dict.items as per python 3.6
+     */
     this.size = 0;
-    this.buckets = {};
+    this.entries = Object.create(null); // avoid clashes with Object.prototype
```

Comment:
doesn't matter _massively_, but `{__proto__:null}` saves you two lookups and a call.

Reply:
i got nervous about that because https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto

Reply:
That's reasonable, it's a legacy feature (albeit one that is unlikely to ever get removed) and `Object.create(null)` is close enough to just as good.